Vungle/VungleRouter.m

```diff
@@ -46,8 +47,11 @@ @interface VungleRouter ()
 @property (nonatomic) SDKInitializeState sdkInitializeState;
 
 @property (nonatomic) NSMutableDictionary *waitingListDict;
+@property (nonatomic) NSMutableDictionary *hbWaitingListDict;
 @property (nonatomic) NSMapTable<NSString *, id<VungleRouterDelegate>> *delegatesDict;
+@property (nonatomic) NSMapTable<NSString *, id<VungleRouterDelegate>> *hbDelegatesDict;
 @property (nonatomic) NSMapTable<NSString *, id<VungleRouterDelegate>> *bannerDelegates;
+@property (nonatomic) NSMapTable<NSString *, id<VungleRouterDelegate>> *hbBannerDelegates;
 
```

Comment:
The key for the original dictionaries is the placement id. The key for the header bidding dictionaries is the event id from the ad markup. The event id and placement id would very unlikely, if ever, be the same or similar.

I think that the waiting lists can be merged into 1 dictionary and all the delegate dictionaries can be merged into 2 map tables (full screen + banner), if not 1 (just a single table for all delegates). Merging the dictionaries would help with object management during development. 

I don't think there is a benefit to split the ads loaded with header bidding.
The only benefit of splitting banner from the full screen delegates is that there doesn't need to be a check if getBannerSize is implemented.

Reply:
The waiting list is a good example. The waiting list contains both full screen and banner delegates if the ad is loaded prior to the SDK being completely initialized.